records/generator.go

```diff
@@ -363,78 +363,100 @@ func (rg *RecordGenerator) listenerRecord(listener string, ns string) {
 
 func (rg *RecordGenerator) taskRecords(sj state.State, domain string, spec labels.Func, ipSources []string) {
 	for _, f := range sj.Frameworks {
-		fname := labels.DomainFrag(f.Name, labels.Sep, spec)
 
-		// insert taks records
-		tail := "." + domain + "."
 		for _, task := range f.Tasks {
 			var ok bool
 			task.SlaveIP, ok = rg.SlaveIPs[task.SlaveID]
 
-			// skip not running or not discoverable tasks
-			if !ok || (task.State != "TASK_RUNNING") {
-				continue
+			// only do running and discoverable tasks
+			if ok && (task.State == "TASK_RUNNING") {
+				rg.taskRecord(task, f, domain, spec, ipSources)
 			}
+		}
+	}
+}
 
-			// define context
-			ctx := struct{ taskName, taskID, slaveID, taskIP, slaveIP string }{
-				spec(task.Name),
-				hashString(task.ID),
-				slaveIDTail(task.SlaveID),
-				task.IP(ipSources...),
-				task.SlaveIP,
-			}
+type context struct {
+	taskName,
+	taskID,
+	slaveID,
+	taskIP,
+	slaveIP string
+}
 
-			// use DiscoveryInfo name if defined instead of task name
-			if task.HasDiscoveryInfo() {
-				ctx.taskName = task.DiscoveryInfo.Name
-			}
+func (rg *RecordGenerator) taskRecord(task state.Task, f state.Framework, domain string, spec labels.Func, ipSources []string) {
 
-			// insert canonical A records
-			canonical := ctx.taskName + "-" + ctx.taskID + "-" + ctx.slaveID + "." + fname
-			arec := ctx.taskName + "." + fname
+	// define context
+	ctx := context{
+		spec(task.Name),
+		hashString(task.ID),
+		slaveIDTail(task.SlaveID),
+		task.IP(ipSources...),
+		task.SlaveIP,
+	}
 
-			rg.insertRR(arec+tail, ctx.taskIP, "A")
-			rg.insertRR(canonical+tail, ctx.taskIP, "A")
+	// use DiscoveryInfo name if defined instead of task name
+	if task.HasDiscoveryInfo() {
+		// LEGACY TODO: REMOVE
+		rg.taskContextRecord(ctx, task, f, domain, spec)
+		// LEGACY, TODO: REMOVE
 
-			rg.insertRR(arec+".slave"+tail, ctx.slaveIP, "A")
-			rg.insertRR(canonical+".slave"+tail, ctx.slaveIP, "A")
+		ctx.taskName = spec(task.DiscoveryInfo.Name)
+		rg.taskContextRecord(ctx, task, f, domain, spec)
+	} else {
+		rg.taskContextRecord(ctx, task, f, domain, spec)
+	}
 
-			// Add RFC 2782 SRV records
-			slaveHost := canonical + ".slave" + tail
-			tcpName := "_" + ctx.taskName + "._tcp." + fname
-			udpName := "_" + ctx.taskName + "._udp." + fname
-			for _, port := range task.Ports() {
-				slaveTarget := slaveHost + ":" + port
+}
+func (rg *RecordGenerator) taskContextRecord(ctx context, task state.Task, f state.Framework, domain string, spec labels.Func) {
+	fname := labels.DomainFrag(f.Name, labels.Sep, spec)
 
-				if !task.HasDiscoveryInfo() {
-					rg.insertRR(tcpName+tail, slaveTarget, "SRV")
-					rg.insertRR(udpName+tail, slaveTarget, "SRV")
-				}
+	tail := "." + domain + "."
 
```

Comment:
Github Diff formatting seems to have brokens with a combination of whitespace changes + reorganization of functions.